src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -287,7 +295,24 @@ internal static string GenerateDataFieldName(ImmutableArray<byte> data)
                 c[i++] = Hexchar(b & 0xF);
             }
 
-            return new string(c);
+            // For alignment of 1 (which is used in cases other than in fields for ReadOnlySpan<byte>),
+            // just use the hex value of the data hash.  For other alignments, tack on a '2', '4', or '8'
+            // accordingly.  As every byte will yield two chars, the odd number of chars used for 2/4/8
+            // alignments will never produce a name that conflicts with names for an alignment of 1.
+            Debug.Assert(alignment is 1 or 2 or 4 or 8);
```

Comment:
Does this assertion hold? Can't `alignment` be `3` here?
If the assertion holds, then it could be tightened by handling the `1` case explicitly and throwing unreachable exception for the switch's default case below.

Reply:
nit: Consider handling the `1` case explicitly in `switch` below and throwing unreachable exception for the switch's default case. Then we don't need this assertion.